Deleted update functions, getSingleNodeRootHash internal functions, a…

Michael-Vander-Meiden opened the pull request:
**Description**
packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol:
- deleted update function
- deleted getSingleNodeRootHash internal function

packages/contracts/contracts/libraries/trie/Lib_SecureMerkleTrie.sol:
- deleted update function
- deleted getSingleNodeRootHash internal function

packages/contracts/test/contracts/libraries/trie/Lib_MerkleTrie.spec.ts
- deleted test for update function

packages/contracts/test/contracts/libraries/trie/Lib_SecureMerkleTrie.spec.ts
- deleted test for update function
- deleted test for getSingleNodeRootHash internal function

**Metadata**
- Fixes #2377

tynes commented:
Can you add a changeset for the contracts package and squash into a single commit? Thanks!

Michael-Vander-Meiden commented:
> 

Sure thing, will do that now

Michael-Vander-Meiden commented:
@tynes done!

tynes commented:
Thanks for updating @Michael-Vander-Meiden! I apologize and should have mentioned that this should be a patch changeset. Pre 1.0.0 we aren't following semver exactly, do you mind updating this? Then it will be good to go

Michael-Vander-Meiden commented:
> Thanks for updating @Michael-Vander-Meiden! I apologize and should have mentioned that this should be a patch changeset. Pre 1.0.0 we aren't following semver exactly, do you mind updating this? Then it will be good to go

Sorry I just saw this. Third time's the charm! Switched it to patch. 😁

tynes commented:
cc @maurelian

maurelian commented on packages/contracts/contracts/libraries/trie/Lib_MerkleTrie.sol:
```diff
@@ -78,35 +78,6 @@ library Lib_MerkleTrie {
-        (uint256 pathLength, bytes memory keyRemainder, ) = _walkNodePath(proof, _key, _root);
-        TrieNode[] memory newPath = _getNewPath(proof, pathLength, _key, keyRemainder, _value);
-
-        return _getUpdatedTrieRoot(newPath, _key);
```
You could also delete this function's implementation (and any subsequent functions called only by it).

Michael-Vander-Meiden replied:
done

maurelian reviewed (approved):
This is safe to merge once the checks are running, but it would be preferable to also delete `_getUpdatedTrieRoot()`.
If we can get that in quickly, great. But I'll approve regardless.

mslipper commented:
Hi Michael!

To get builds running please unfollow your fork of `optimism` and follow the repo under our organization. CircleCI uses which repo you follow to determine which repo to build. More details from CircleCI's documentation:

> Note: If a user submits a pull request to your repository from a fork, but no pipeline is triggered, then the user most likely is following a project fork on their personal account rather than the project itself of CircleCi, causing the jobs to trigger under the user’s personal account and not the organization account. To resolve this issue, have the user unfollow their fork of the project on CircleCI and instead follow the source project. This will trigger their jobs to run under the organization when they submit pull requests.

maurelian commented:
Ugh, sorry for the trouble @Michael-Vander-Meiden, but you'll need to rebase this on `develop`, and run `yarn lint` to [fix that failed check](https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/1359/workflows/96d3ef0e-4bff-48e2-a92f-b922a33a88c5/jobs/4555/parallel-runs/0/steps/0-103).

Thanks for working through this with us.

Michael-Vander-Meiden commented:
No worries, working on it now.

Michael-Vander-Meiden commented:
hey @maurelian , just want to make sure I get this right. When you say rebase on `develop`, do you mean pull `develop` into my `iss2377` branch and then lint, or something else?

maurelian commented:
`git pull --rebase origin develop` should be the command you need. The important thing is that your changes are made on top of the latest commit on our `develop`.

Michael-Vander-Meiden commented:
Hey @maurelian ! There ended up being *a lot* more unused functions than I originally thought. I deleted those as well some more tests and fixed the linting issues.

If things still look good to you then I will go ahead and squash these commits to prep for merge.

maurelian commented:
Nothing like the smell of deleted code in the morning!These changes look great, squash away.

Michael-Vander-Meiden commented:
@maurelian looks ready for merge!😁
